fabric/src/test/java/TestRegexs.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+import com.wynntils.features.chat.GuildRankReplacementFeature;
+import com.wynntils.features.ui.BulkBuyFeature;
+import com.wynntils.models.items.annotators.gui.ArchetypeAbilitiesAnnotator;
+import java.lang.reflect.Field;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TestRegexs {
+    @Test
+    public void testArchetypeName() {
+        Pattern pattern =
+                TestPrivateFieldsUtil.getPrivateStaticRegexPattern(ArchetypeAbilitiesAnnotator.class, "ARCHETYPE_NAME");
+        assertMatchesAny(pattern, "§e§lBoltslinger Archetype").assertGroupIs(1, "e");
+        assertMatchesAny(pattern, "§d§lSharpshooter Archetype").assertGroupIs(1, "d");
+        assertMatchesAny(pattern, "§2§lTrapper Archetype").assertGroupIs(1, "2");
+        assertMatchesAny(pattern, "§d§lLight Bender Archetype").assertGroupIs(1, "d");
+    }
+
+    @Test
+    public void testArchetypePattern() {
+        Pattern pattern = TestPrivateFieldsUtil.getPrivateStaticRegexPattern(
+                ArchetypeAbilitiesAnnotator.class, "ARCHETYPE_PATTERN");
+        assertMatchesAny(pattern, "§a✔ §7Unlocked Abilities: §f14§7/15")
+                .assertGroupIs(1, "14")
+                .assertGroupIs(2, "15");
+        assertMatchesAny(pattern, "§a✔ §7Unlocked Abilities: §f2§7/16")
+                .assertGroupIs(1, "2")
+                .assertGroupIs(2, "16");
+        assertMatchesAny(pattern, "§a✔ §7Unlocked Abilities: §f0§7/15")
+                .assertGroupIs(1, "0")
+                .assertGroupIs(2, "15");
+    }
+
+    @Test
+    public void testPricePattern() {
+        Pattern pattern = TestPrivateFieldsUtil.getPrivateStaticRegexPattern(BulkBuyFeature.class, "PRICE_PATTERN");
+        assertMatchesAny(pattern, "§6 - §a✔ §f10§7²");
+        assertMatchesAny(pattern, "§6 - §a✔ §f1483§7²");
+        assertMatchesAny(pattern, "§6 - §c✖ §f244§7²");
+        assertMatchesAny(pattern, "§6 - §c✖ §f1§7²");
+        assertMatchesAny(pattern, "§6 - §a✔ §f24§7²");
+    }
+
+    @Test
+    public void testGuildMessagePattern() {
+        Pattern pattern = TestPrivateFieldsUtil.getPrivateStaticRegexPattern(
```

Comment:
This still looks a bit clunky, but I'm honestly not sure how to make it smoother. 

Maybe as a builder? Like
```
new RegExTester( GuildRankReplacementFeature.GUILD_MESSAGE_PATTERN)
.matching("§3[§b★★★★★§3§oDisco reroller§3]")
.matching("§3[§b★★★★§3§obol§r§3] ")
.notMatching("banana pie with cream")
.test();
```

Reply:
The point is that it should be an absolute minimal amount of boilerplate, so the actual test strings really shine through the code framework.

Reply:
Also, as can be seen in this example, I'm not 100% sure I like the reflection lookup of the fields. I think it is better to make them public so the test can access them. The point in doing that is that you will get IntelliJ assist in going from the actual regex to the test and vice versa, by the "usage" linking.

But, this also breaks encapsulation, so I'm slightly at the fence. What do you say, @kristofbolyai?

Reply:
Hm, I don't like either version. Let's think about it a but more later on

Reply:
Ideally, we'd be able to expose the field to the test, and the test only. I don't think it is possible in Java, though. Maybe some `@annotations` on the field could help? But I think not.

Reply:
there is `@TestOnly` or `@VisibleForTesting`. VisibleForTesting is exactly what we need but it is only since 20.0.0 according to the docs:

> A member or type annotated with VisibleForTesting claims that its visibility is higher than necessary, only for testing purposes. In particular:
> If public or protected member/type is annotated with VisibleForTesting, it's assumed that package-private access is enough for production code.
> If package-private member/type is annotated with VisibleForTesting, it's assumed that private access is enough for production code.
> It's illegal to annotate private member/type as VisibleForTesting.
> This annotation means that the annotated element exposes internal data and breaks encapsulation of the containing class; the annotation won't prevent its use from production code, developers even won't see warnings if their IDE doesn't support the annotation. It's better to provide proper API which can be used in production as well as in tests.
> Since:
> 20.0.0

Reply:
So it doesn't do anything

Reply:
"Go to Preferences | Editor | Inspections and enable the inspection Java | General | Test-only class or method call in production code. Now you will get a warning if you try to call members with the @VisibleForTesting annotation from production code, but not from test code." - StackOverflow
isn't that what we wanted?

Reply:
so like.... now what?

Reply:
Well, in the real world, you don't unit tests fields. You unit test behavior. Behavior of the model. But here, we want to bend the rules, which is understandable. But test suites are not made for this. So we will always need to compromise something. Code quality, testing "usefulness" or both.